file_writer_control/InThreadStatusTracker.py

```diff
@@ -0,0 +1,212 @@
+from streaming_data_types.finished_writing_wrdn import (
+    FILE_IDENTIFIER as STOPPED_IDENTIFIER,
+)
+from datetime import datetime, timedelta
+from streaming_data_types.utils import _get_schema
+from streaming_data_types import deserialise_x5f2 as deserialise_status
+from streaming_data_types import deserialise_answ as deserialise_answer
+from streaming_data_types import deserialise_6s4t as deserialise_stop_time
+from streaming_data_types import deserialise_pl72 as deserialise_start
+from streaming_data_types import deserialise_wrdn as deserialise_stopped
+from streaming_data_types.run_stop_6s4t import RunStopInfo
+from streaming_data_types.run_start_pl72 import RunStartInfo
+from streaming_data_types.status_x5f2 import StatusMessage
+from streaming_data_types.finished_writing_wrdn import WritingFinished
+from streaming_data_types.action_response_answ import FILE_IDENTIFIER as ANSW_IDENTIFIER
+from file_writer_control.JobStatus import JobStatus, JobState
+from file_writer_control.CommandStatus import CommandStatus, CommandState
+from file_writer_control.WorkerStatus import WorkerState, WorkerStatus
+from file_writer_control.StateExtractor import (
+    extract_worker_state_from_status,
+    extract_state_from_command_answer,
+    extract_job_state_from_answer,
+)
+import json
+from streaming_data_types.action_response_answ import Response
+from typing import Dict
+
+STATUS_MESSAGE_TIMEOUT = timedelta(seconds=5)
+JOB_STATUS_TIMEOUT = timedelta(seconds=5)
+COMMAND_STATUS_TIMEOUT = timedelta(seconds=20)
+
+
+class InThreadStatusTracker:
+    """
+    Implements de-coding of flatbuffer messages and sends updates of worker, job and command state/status back to the
+    "main"-thread if there has been changes.
+    """
+
+    def __init__(self, status_queue: Queue):
+        """
+        Constructor.
+        :param status_queue: The output queue to which state/status updates are pushed.
+        """
+        self.queue = status_queue
+        self.known_workers: Dict[str, WorkerStatus] = {}
+        self.known_jobs: Dict[str, JobStatus] = {}
+        self.known_commands: Dict[str, CommandStatus] = {}
+
+    def process_message(self, message: bytes):
+        """
+        Process a binary message.
+        :param message: The binary message to be processed.
+        """
+        current_schema = _get_schema(message).encode("utf-8")
+        update_time = datetime.now()
+        if current_schema == ANSW_IDENTIFIER:
+            self.process_answer(deserialise_answer(message))
+        elif current_schema == STAT_IDENTIFIER:
+            self.process_status(deserialise_status(message))
+        elif current_schema == STOP_TIME_IDENTIFIER:
+            self.process_set_stop_time(deserialise_stop_time(message))
+        elif current_schema == START_IDENTIFIER:
+            self.process_start(deserialise_start(message))
+        elif current_schema == STOPPED_IDENTIFIER:
+            self.process_stopped(deserialise_stopped(message))
```

Comment:
While not critical, if-elif sequences can be shorthanded via using Python dictionaries; i.e., while case-switch structure does not exist in Python, it is very easy to create. For example [see here](https://uwpce-pythoncert.github.io/PythonCertDevel/modules/DictionaryAsSwitch.html) which is very relevant to this code.

Reply:
Doing it this way requires the use of functions or lambdas which then in turn results in the same number of lines. Still, I have re-written it now so I might as well leave it as is.

Reply:
My intention was not to reduce the number of lines but rather to have a more Pythonic way. The current code with dictionary is certainly more elegant! Great job!